build-logic/src/main/java/qsl/internal/extension/QslModuleExtensionImpl.java

```diff
@@ -0,0 +1,180 @@
+	private static final long serialVersionUID = 2L;
+	// public properties
+	@Input
+	private final Property<String> name;
+	@Input
+	private final Property<String> library;
+	@Input
+	private final Property<String> moduleName;
+	@Input
+	private final Property<String> id;
+	@Input
+	private final Property<String> description;
+	@Input
+	private final Property<Environment> environment;
+	@Input
+	private final Property<Boolean> hasAccessWidener;
+	@Input
+	private final Property<Boolean> hasMixins;
+	@Nested
+	private final NamedDomainObjectContainer<QslLibraryDependency> moduleDependencyDefinitions;
+	@Nested
+	private final NamedDomainObjectContainer<EntrypointObjectHolder> entrypoints;
+
+
+
+	@Inject
+	public QslModuleExtensionImpl(ObjectFactory factory, Project project) {
+		super(project);
+		this.library = factory.property(String.class);
+		this.library.finalizeValueOnRead();
+		this.moduleName = factory.property(String.class);
+		this.moduleName.finalizeValueOnRead();
+		this.id = factory.property(String.class);
+		this.id.finalizeValueOnRead();
+		this.name = factory.property(String.class);
+		this.name.finalizeValueOnRead();
+		this.description = factory.property(String.class);
+		this.description.finalizeValueOnRead();
+		this.environment = factory.property(Environment.class);
+		this.environment.set(Environment.ANY);
+		this.environment.finalizeValueOnRead();
+		this.hasAccessWidener = factory.property(Boolean.class);
+		this.hasAccessWidener.set(false);
+		this.hasAccessWidener.finalizeValueOnRead();
+		this.hasMixins = factory.property(Boolean.class);
+		hasMixins.set(true);
+		this.hasMixins.finalizeValueOnRead();
+		this.entrypoints = factory.domainObjectContainer(EntrypointObjectHolder.class, n -> new EntrypointObjectHolder(factory, n));
+		//this.moduleDependencies = new ArrayList<>();
+		LicenseHeader licenseHeader = new LicenseHeader(
+				LicenseHeader.Rule.fromFile(project.getRootProject().file("codeformat/FABRIC_MODIFIED_HEADER").toPath()),
+				LicenseHeader.Rule.fromFile(project.getRootProject().file("codeformat/HEADER").toPath())
+		);
+
+		this.moduleDependencyDefinitions = factory.domainObjectContainer(QslLibraryDependency.class, name -> new QslLibraryDependency(factory, name));
+
+		project.getTasks().register("checkLicenses", CheckLicenseTask.class, licenseHeader);
+		project.getTasks().register("applyLicenses", ApplyLicenseTask.class, licenseHeader);
+		project.getTasks().findByName("check").dependsOn("checkLicenses");
+	}
+
+	public Property<String> getModuleName() {
+		return this.moduleName;
+	}
+
+	public Property<String> getLibrary() {
+		return this.library;
+	}
+
+	public Property<String> getName() {
+		return this.name;
+	}
+	public Property<String> getId() {
+		return this.id;
+	}
+	public Property<String> getDescription() {
+		return this.description;
+	}
+
+	public Property<Boolean> getHasAccessWidener() {
+		return hasAccessWidener;
+	}
+
+	public void accessWidener() {
```

Comment:
This doesnt conflict but does the same thing as the access widener method above. I would pick one. (Also the suggestion for the AW name should be there as well)